src/engine/keycontrol.cpp

```diff
@@ -212,12 +217,16 @@ void KeyControl::updateRate() {
         // !bKeylock
         if (m_pitchRateInfo.keylock) {
             // Disabling Keylock
-            if (m_keylockMode->get() == kLockCurrentKey) {
-                // reset to linear pitch
-                m_pitchRateInfo.pitchTweakRatio = 1.0;
-                // For not resetting to linear pitch:
-                // Adopt speedPitchRatio change as pitchTweakRatio
-                //pitchRateInfo.pitchTweakRatio *= (m_speedSliderPitchRatio / pitchRateInfo.tempoRatio);
+            // If "Reset (unlocked) Key" is enabled
+            if (m_keyunlockMode->get() == kResetUnlockedKey) {
+                // If 'current' aka 'not original' key was locked
+                if (m_keylockMode->get() == kLockCurrentKey) {
```

Comment:
Regardless of my changes, I wonder what happens here if original key was locked but changed afterwards, manually or with "match key"